File: 백준/Silver/2606. 바이러스/바이러스.py
N = int(input()) # 컴퓨터의 수
M = int(input()) # 컴퓨터 연결선 수

graph = [[] for _ in range(N + 1)] # 그래프
visited = [0] * (N + 1)

# 컴퓨터 연결 그래프 초기화
for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
    graph[a].sort()

def dfs(num):
    visited[num] = 1 # 방문처리

    for i in graph[num]:
        if visited[i] == 0:
            dfs(i)
    return

# 1번 컴퓨터
dfs(1) # graph = [[], [2, 5], [1, 3, 5], [2], [7], [1, 2, 6], [5], [4]]

# 감염된 컴퓨터 개수 계산
result = 0
for visit in visited:
    result += visit

print(result - 1) # 1번 컴퓨터는 개수에서 빼준다

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;

class Main {

    static int[][] graph;
    static boolean[] visited;
    static int answer = 0;
    static int N;
    static int M;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = Integer.parseInt(sc.nextLine());
        M = Integer.parseInt(sc.nextLine());

        visited = new boolean[N + 1];
        graph = new int[N + 1][N + 1];
        for (int i = 0; i < M; i++) {
            String[] s = sc.nextLine().split(" ");
            int a = Integer.parseInt(s[0]);
            int b = Integer.parseInt(s[1]);
            graph[a][b] = graph[b][a] = 1;
        }
        visited[1] = true;
        BFS(1);
        System.out.println(answer);
    }

    static void BFS(int v) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(v);
        while (!q.isEmpty()) {
            Integer nowV = q.poll();
            for (int i = 1; i < N + 1; i++) {
                if (graph[nowV][i] == 1 && !visited[i]) {
                    answer++;
                    visited[i] = true;
                    q.offer(i);
                }
            }
        }
    }
}